pkg/services/secrets/database/database.go

```diff
@@ -75,20 +81,20 @@ func (ss *SecretsStoreImpl) GetCurrentDataKey(ctx context.Context, label string)
 
 func (ss *SecretsStoreImpl) GetAllDataKeys(ctx context.Context) ([]*secrets.DataKey, error) {
 	result := make([]*secrets.DataKey, 0)
-	err := ss.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
+	err := ss.db.WithDbSession(ctx, func(sess *db.Session) error {
 		err := sess.Table(dataKeysTable).Find(&result)
 		return err
 	})
 	return result, err
 }
 
-func (ss *SecretsStoreImpl) CreateDataKey(ctx context.Context, dataKey *secrets.DataKey) error {
-	return ss.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
-		return ss.CreateDataKeyWithDBSession(ctx, dataKey, sess.Session)
+func (ss *SecretsStoreImpl) CreateDataKey(ctx context.Context, dataKey *secrets.DataKey, fns ...secrets.OnSuccessfulDataKeyCreation) error {
+	return ss.db.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
+		return ss.createDataKeyWithDBSession(ctx, dataKey, sess, fns...)
 	})
 }
 
-func (ss *SecretsStoreImpl) CreateDataKeyWithDBSession(_ context.Context, dataKey *secrets.DataKey, sess *xorm.Session) error {
+func (ss *SecretsStoreImpl) createDataKeyWithDBSession(_ context.Context, dataKey *secrets.DataKey, sess *db.Session, fns ...secrets.OnSuccessfulDataKeyCreation) error {
```

Comment:
nit: probably no need to keep it a separate method any more and it's body can be just moved to `CreateDataKey` instead?